dev/RatingControl/RatingControl.cpp

```diff
@@ -46,6 +47,34 @@ float RatingControl::ActualRatingFontSize()
     return RenderingRatingFontSize() / 2;
 }
 
+double RatingControl::ItemSpacing()
+{
+    // TextScaleFactor will change font size but won't affect spacing/margin.
+    // The scaled-up glyph will grow out to the margin area, so the "actual" margins become smaller.
+    // Therefore we should include TextScaleFactor when calculating item spacing in order to get correct total width and star center positions.
+    return c_defaultItemSpacing / GetUISettings().TextScaleFactor();
+}
```

Comment:
I think this should be an inline function. Per the F.5 rule [http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-inline](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-inline)

Reply:
Inline functions should go in the header file.

---
In reply to: [248481868](https://github.com/Microsoft/microsoft-ui-xaml/pull/187#discussion_r248481868) [](ancestors = 248481868)

Reply:
We don't know this is a time-critical function. I would rather let the compiler decide and leave the code in the cpp file.

Reply:
Yeah it's probably not time critical, I'll remove it for now then.